cmds/ls/fileinfo_linux.go

```diff
@@ -4,153 +4,16 @@
 
 package main
 
-import (
-	"fmt"
-	"os"
-	"os/user"
-	"regexp"
-	"strconv"
-	"strings"
-	"syscall"
-	"time"
-
-	humanize "github.com/dustin/go-humanize"
-)
-
 // From Linux header: /include/uapi/linux/kdev_t.h
```

Comment:
I dug a bit deeper into how the how linux device nodes are encoded. The new (since kernel 2.6) encoding supports 12 bits for the major and 20 bits for the minor with these masks:

	unsigned major = (dev & 0xfff00) >> 8;
	unsigned minor = (dev & 0xff) | ((dev >> 12) & 0xfff00);

https://elixir.bootlin.com/linux/latest/source/include/linux/kdev_t.h#L46

This encoding is backwards-compatible.